akka-projection-cassandra/src/it/scala/akka/projection/cassandra/CassandraProjectionSpec.scala

```diff
@@ -426,13 +430,17 @@ class CassandraProjectionSpec
           sourceProbe.get.sendNext(Envelope(entityId, n, s"elem-$n"))
         }
         offsetShouldBe(projectionId, 10L)
-        repository.findById(entityId).futureValue.get.text should include("elem-15")
+        eventually {
+          repository.findById(entityId).futureValue.get.text should include("elem-15")
+        }
 
         (16 to 17).foreach { n =>
           sourceProbe.get.sendNext(Envelope(entityId, n, s"elem-$n"))
         }
         offsetShouldBe(projectionId, 17L)
-        repository.findById(entityId).futureValue.get.text should include("elem-17")
+        eventually {
+          repository.findById(entityId).futureValue.get.text should include("elem-17")
+        }
```

Comment:
Eventually because it timed out or why?

Reply:
Also interested in the reason. Since it's atLeastOnce the repository should be updated first and the offset saved afterwards. Therefore I don't see a reason for eventually.

Reply:
The `findById` did not return all elements:
```
<-- [akka.projection.cassandra.CassandraProjectionSpec: A Cassandra at-least-once projection must save offset after idle duration] End of log messages of test that [Failed(org.scalatest.exceptions.TestFailedException: "elem-1|elem-2|elem-3|elem-4|elem-5|elem-6|elem-7|elem-8|elem-9|elem-10|elem-11" did not include substring "elem-15")]
[info] - must save offset after idle duration *** FAILED *** (135 milliseconds)
[info]   "elem-1|elem-2|elem-3|elem-4|elem-5|elem-6|elem-7|elem-8|elem-9|elem-10|elem-11" did not include substring "elem-15" (CassandraProjectionSpec.scala:429)
```

Note if fails a bit randomly, so the log above is from a different test. But eventually, this one we are commenting on will also fail.

Reply:
Aha, so it's about when findById is called vs updates complete. Then I think eventually is a fine test improvement.

Reply:
Seems so. @patriknw Anything from your end? Otherwise I can get it merged as its needed for the scala3 build.

Reply:
In the line above we have verified that the stored offset is 17. The repository storage of elem-17 should happen before offset store of 17, otherwise it would not be at-least-once. If the update has completed the findById should also see that value, as far as I know.

I still don't understand this, and [we have several fail issues for CassandraSpec](https://github.com/akka/akka-projection/issues?q=is%3Aissue+is%3Aopen+CassandraProjectionSpec) so I suggest that we create a new issue for this and revert this from this PR - unless you can explain what is wrong with my reasoning.

Reply:
I missed the offset store read, indeed very strange, like the projection write completes but cannot be observed until after a while.

Reply:
Interesting, since only the version of `scalatest` / `scalatestplus` has changed.

Reply:
The Cassandra tests are flaky since before, and that's why I want us to fix properly and not paper over.

Reply:
Actually, it's only eventually for this elem-17 that should be reverted. The others above should have eventually because there we don't check for the corresponding offset first 10 vs elem-15. The test failure you mentioned above was for "elem-15"